documentation/remote-host-object.md

```diff
@@ -0,0 +1,27 @@
+# Support for remote host object
+
+Remote host object need to be registered in [Running Object Table(ROT)](https://docs.microsoft.com/en-us/windows/desktop/api/objidl/nn-objidl-irunningobjecttable) before hand. And msbuild will call [IRunningObjectTable::GetObject](https://docs.microsoft.com/en-us/windows/desktop/api/objidl/nf-objidl-irunningobjecttable-getobject) to get the host object by monikerName which is passed in method `void RegisterHostObject(string projectFile, string targetName, string taskName, string monikerName)`.
+
+[The registration of interfaces](https://docs.microsoft.com/en-us/dotnet/framework/interop/how-to-register-primary-interop-assemblies) is the only thing interop with COM that need extra care. Registration could be global with register or without. Other interface needed is registered globally by Visual Studio during installation. Ony `ITaskHost` does not use global registration. On VS side, the proxy is written. But on MsBuild side it was not. That result the change in msbuild.exe.manifest file and the change to generate tlb file for ITaskHost.
+
+## Change in  msbuild.exe.manifest file.
+```
+<file name="Microsoft.Build.Framework.tlb"> --Location of the tlb, it should be in the same directory of msbuild.exe
```

Comment:
This raises an interesting point: **do we need to support this for amd64 MSBuild?**

Currently you're placing the `.tlb` only in the 32-bit MSBuild location, but VS can (if the appropriate hoops are jumped through) execute 64-bit tasks.

Reply:
For this work only 32 bit is needed. From @davkean

Reply:
Awesome. We should probably make that explicit -- if a host object is registered by moniker and accessed in amd64 MSBuild, throw?